common/corerequestcontext.go

```diff
@@ -88,10 +88,15 @@ func UpdateResponseStatus(ctx context.Context, status int) error {
 	return nil
 }
 
-func CoreRequestContextMiddleware() func(next http.Handler) http.Handler {
+func CoreRequestContextMiddleware(logger *logrus.Logger) func(next http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		fn := func(w http.ResponseWriter, r *http.Request) {
-			ctx := log.With(traceIDLogField, GetTraceIDFromContext(r.Context())).Onto(r.Context())
+			ctx := r.Context()
+			// insert the logger to the context if the logger is passed in and no existing logger exists
+			if logger != nil && getCoreContext(ctx).logger != nil {
+				ctx = LoggerToContext(ctx, logger, nil)
+			}
```

Comment:
This could simply be done once before this func setup as its either always going to be needed or never